src/app/Providers/AppServiceProvider.php

```diff
@@ -23,6 +25,37 @@ public function register()
      */
     public function boot()
     {
-        //
+        if (ENV('LOG_DATABASE_QUERIES',
+            default: false)) {
+            $this->logDatabaseQueries();
+        }
+
     }
+
+    private function logDatabaseQueries(
+    )
+    {
+        DB::listen(function (
+            $query
+        ) {
+
+            $routeName = optional(request()->route())->getName();
+            $url = request()->fullUrl();
+
+            // Log des informations
+            Log::info('Database Query Executed',
+                [
+                    'tag' => 'database_event',
+                    'type' => 'query_executed',
+                    'sql' => $query->sql,
+                    'time_ms' => $query->time,
+                    'route_name' => $routeName ?? 'unknown',
+                    'url' => $url,
+                    'request_id' => request()->header('X-Request-ID',
+                        'unknown'),
+                ]);
```

Comment:
Do we also want to add `user_id` and `crew_id`?

Reply:
For some reason the app crashes when logging it, so I just remove it

Reply:
Okay I just tried and it crashes as well. Not a big deal since we have the ids on the model_events with the same request_id

Reply:
I didn't have the time to check why it crashes. But yep, you're right we can get it throught the request_id